### PR TITLE
[Feature] Prevent Autoscaling of UIKitComponent (RadioButtonGroup).

### PR DESCRIPTION
**Pull request by michael-zimmermann**
POC: 
I've added a bool to disable / enable autoscaling and used the radio button group (UIKit) as an example.

I'd welcome some feedback

### CODE REVIEW

File: core/Sources/Theming/Content/Typography/Typography.swift [hunk 1]
```diff
@@ -41,4 +41,11 @@ public protocol Typography {
 public protocol TypographyFontToken {
     var uiFont: UIFont { get }
     var font: Font { get }
+    func uiFont(isScaled: Bool) -> UIFont
```
Comment (LouisBorleeAdevinta) [inline core/Sources/Theming/Content/Typography/Typography.swift:44]: I think this should be internal, not public

**Review by LouisBorleeAdevinta — COMMENTED** [review]
Hey that's great! It's pretty much what I had in mind :) 
I was thinking defaultValue to false for everybody
I was wondering if we could make the var public but I don't think it's a good idea or even necessary

File: core/Sources/Components/RadioButton/View/UIKit/RadioButtonUIGroupViewTests.swift [hunk 2]
```diff
@@ -126,7 +130,8 @@ final class RadioButtonUIGroupViewTests: UIKitComponentTestCase {
             selectedID: self.selectedID,
             items: self.items,
             radioButtonLabelPosition: .left,
-            groupLayout: .vertical
+            groupLayout: .vertical,
+            isAutoscalingEnabled: true
         )
```
Comment (LouisBorleeAdevinta) [inline core/Sources/Components/RadioButton/View/UIKit/RadioButtonUIGroupViewTests.swift:136]: Are there tests to verify it doesn't scale?

File: core/Sources/Theming/Content/Typography/Typography.swift (hunk 1 above)
Comment (LouisBorleeAdevinta) [reply core/Sources/Theming/Content/Typography/Typography.swift:44]: Shouldn't it be removed from here?

File: core/Sources/Theming/Content/Typography/TypographyDefault.swift [hunk 3]
```diff
@@ -96,6 +96,18 @@ public struct TypographyFontTokenDefault: TypographyFontToken {
                            relativeTo: textStyle)
     }
 
+    public func uiFont(isScaled: Bool = false) -> UIFont {
```
Comment (LouisBorleeAdevinta) [inline core/Sources/Theming/Content/Typography/TypographyDefault.swift:99]: I don't think it ought to be used by consumers, should be internal

File: core/Sources/Theming/Content/Typography/Typography.swift (hunk 1 above)
Comment (michael-zimmermann) [reply core/Sources/Theming/Content/Typography/Typography.swift:44]: The problem is if I remove it from here, then I can only have a default implementation, but can't overwrite the default implementation.
If I have it as part of the protocol, I can write a default implementation in the extension and overwrite the default implementation.

File: core/Sources/Extension/PropertyWrapper/ScaledUIMetric.swift [hunk 4]
```diff
@@ -12,10 +12,12 @@ import UIKit
 @propertyWrapper struct ScaledUIMetric<Value> where Value : BinaryFloatingPoint {
     // MARK: - Properties
 
+    let isAutoscalable: Bool
```
Comment (janniklasfreundt) [inline core/Sources/Extension/PropertyWrapper/ScaledUIMetric.swift:15]: Maybe some tests for this in `ScaledUIMetricTests` would be helpful?

**Pull request closed without merging**